### PR TITLE
Borgs pull corpses slightly slower than fullspeed

### LINKED ISSUE #1126: Borg slowed when dragging someone.
[BUG]

**Describe the bug**
When dragging someone, borg are being slowed as is the normal case with humans. I didn't see anything in the changelog so I assume this is a bug?

**To Reproduce**
1. Be borg
2. Drag person alive or dead
3. Be super slowed
4. Be sad

**Expected behavior**
Being able to drag someone at normal speed.

### PR DESCRIPTION
**Pull request by Tarmunora**
[BALANCE] [FIX] [CLEANLINESS]
## About the PR 
Make borgs pull corpses slightly slower than fullspeed (about 20% of standard movement penalty for mob-pulling)

Merge pull_speed_modifier proc into the exactly one place it was used - change pullspeed multipliers to be a linear interpolation b/w 1 and the higher value instead of a direct multiplier
Add a mob_pull_multiplier for movement modifiers that affects pullspeed for pulling mobs seperately from general pullspeed

## Why's this needed? 
closes #1126
Supersedes #1143 

## Changelog

```
(u)Tarmunora:
(*)Borgs will now pull mobs at slightly less than full speed
```

### CODE REVIEW

File: code/mob/living.dm [hunk 1]
```diff
@@ -1502,10 +1504,33 @@ var/global/icon/human_static_base_idiocy_bullshit_crap = icon('icons/mob/human.d
 
 	if (pushpull_multiplier != 0) // if we're not completely ignoring pushing/pulling
 		if (src.pulling)
-			. *= (pull_speed_modifier(move_target) * pushpull_multiplier)
+			if (istype(src.pulling, /atom/movable) && !(src.is_hulk() || (src.bioHolder && src.bioHolder.HasEffect("strong"))))
```
Comment (mordent-goonstation) [inline code/mob/living.dm:1507]: Bringing this in to here for performance (i.e. only used once, so might as well have here)?

File: code/mob/living.dm [hunk 2]
```diff
@@ -1502,10 +1504,33 @@ var/global/icon/human_static_base_idiocy_bullshit_crap = icon('icons/mob/human.d
...
+						else if(istype(A, /obj/storage))
+							// if the storage object contains mobs, use its p_class (updated within storage to reflect containing mobs or not)
+							if (locate(/mob) in A.contents)
+								. *= lerp(1, max(A.p_class, 1), mob_pull_multiplier)
+			. = lerp(1, . , pushpull_multiplier)
+
 
 		if (src.pushing && (src.pulling != src.pushing))
-			. *= (max(src.pushing.p_class, 1) * pushpull_multiplier)
+			. *= lerp(1, max(src.pushing.p_class, 1), pushpull_multiplier)
```
Comment (mordent-goonstation) [inline code/mob/living.dm:1533]: Didn't recognize this proc, looked into it. This really should be a macro. It's used in so few places I wouldn't even be against you doing in this PR, but if you'd rather not I'll probably create an Issue for it.

File: code/datums/movement_modifier/movement_modifiers.dm [hunk 3]
```diff
@@ -79,6 +80,7 @@
 // robot modifiers
 /datum/movement_modifier/robot_base
 	health_deficiency_adjustment = -INFINITY
+	mob_pull_multiplier = 0.2
```
Comment (mordent-goonstation) [inline code/datums/movement_modifier/movement_modifiers.dm:83]: Add a comment to explain this deviation, ideally.

Comment (mordent-goonstation) [inline code/datums/movement_modifier/movement_modifiers.dm:83]: How does this affect the AI's bot's ability to shove people around?

File: code/mob/living.dm (hunk 2 above)
Comment (Tarmunora) [reply code/mob/living.dm:1533]: saner to macroize separately I think

File: code/mob/living.dm (hunk 1 above)
Comment (Tarmunora) [reply code/mob/living.dm:1507]: That and it lets me use the values derived from movement modifiers (such as mob_pull_multiplier)

File: code/datums/movement_modifier/movement_modifiers.dm (hunk 3 above)
Comment (Tarmunora) [reply code/datums/movement_modifier/movement_modifiers.dm:83]: Should only affect specifically cyborgs *pulling* mobs

Comment (Tarmunora) [reply code/datums/movement_modifier/movement_modifiers.dm]: Done

**Comment by Tiggersaurus** [comment]
😍

**Pull request merged**